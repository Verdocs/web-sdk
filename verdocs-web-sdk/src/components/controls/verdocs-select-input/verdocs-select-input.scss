@import '../../../theme.scss';

verdocs-select-input {
  font-family: $primary-font;
  box-sizing: border-box;
  margin: 0 0 10px 0;
  display: block;

  .input-element {
    box-sizing: border-box;
    background: $verdocs-grey-4;
    border: 1px solid $verdocs-bg-1;
    border-radius: 4px;
    background: $verdocs-grey-4;
    color: $verdocs-grey-0;
    width: 100%;
    font-size: 16px;
    height: 28px;
    padding: 2px 8px;

    &::placeholder {
      color: $light-disabled-color;
      opacity: 1;
    }

    &:focus {
      //border: 1px solid $border-color-highlight;
    }
  }

  .input-label {
    display: block;
    color: $label-color-light;
    font-weight: 700;
    font-size: 14px;
    margin: 0 0 4px 0;
  }
}
