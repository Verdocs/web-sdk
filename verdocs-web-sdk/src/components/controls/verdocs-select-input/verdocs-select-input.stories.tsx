import {html} from 'lit-html';
import {Meta} from '@storybook/web-components';

export default {
  title: 'Controls/Select Input',
  component: 'verdocs-select-input',
  args: {
    label: 'Name',
    value: '',
    options: [
      {label: 'Option 1', value: '1'},
      {label: 'Option 2', value: '2'},
      {label: 'Option 3', value: '3'},
    ],
  },
  argTypes: {
    change: {
      action: 'change',
      table: {
        disable: true,
      },
    },
  },
} as Meta;

export const SelectInput = ({type, label, value, placeholder, options, disabled, change}) =>
  html`<verdocs-select-input .type=${type} .label=${label} .placeholder=${placeholder} .value=${value} .options=${options} .disabled=${disabled} @change=${change} />`;
