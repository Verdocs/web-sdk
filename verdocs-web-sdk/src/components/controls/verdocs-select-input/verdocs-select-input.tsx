import {Component, Prop, Host, h} from '@stencil/core';

/**
 * Display a combo box. This is just a standard HTML select field with minimal markup to fit the
 * visual styles of the other components. Note that events "bubble" from the input field to the container,
 * so you can subscribe to the same DOM events (input, blur, etc) that a normal input would emit.
 *
 * ```ts
 * <verdocs-select-input label="Select:" label="Select" options={[...options]} onInput={() => {}} />
 * ```
 */
@Component({
  tag: 'verdocs-select-input',
  styleUrl: 'verdocs-select-input.scss',
})
export class VerdocsSelectInput {
  /**
   * The initial value for the input field.
   */
  @Prop() value: string = '';

  /**
   * The label for the field.
   */
  @Prop() label: string = '';

  /**
   * The options to list.
   */
  @Prop() options: {label: string; value: string}[];

  /**
   * Should the field be disabled?
   */
  @Prop() disabled: boolean = false;

  render() {
    return (
      <Host class="input-field">
        <label>
          {this.label ? <div class="input-label">{this.label + ':'}</div> : <div />}
          <select class="input-element" disabled={this.disabled}>
            {this.options.map(option => (
              <option value={option.value} selected={option.value === this.value}>
                {option.label}
              </option>
            ))}
          </select>
        </label>
      </Host>
    );
  }
}
