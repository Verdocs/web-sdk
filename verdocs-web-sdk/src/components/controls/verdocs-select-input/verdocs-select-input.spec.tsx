import { newSpecPage } from '@stencil/core/testing';
import { VerdocsSelectInput } from './verdocs-select-input';
import { h } from '@stencil/core';

describe('verdocs-select-input', () => {
  it('renders with default props', async () => {
    const page = await newSpecPage({
      components: [VerdocsSelectInput],
      template: () => <verdocs-select-input options={[]} />,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('renders with label, value, and options', async () => {
    const options = [
      { label: 'Option 1', value: '1' },
      { label: 'Option 2', value: '2' }
    ];
    const page = await newSpecPage({
      components: [VerdocsSelectInput],
      template: () => <verdocs-select-input label="Select" value="2" options={options} />,
    });
    expect(page.root).toMatchSnapshot();
    const selectedOption = page.root.querySelector('option[selected]');
    expect(selectedOption).not.toBeNull();
    expect(selectedOption?.getAttribute('value')).toBe('2');
  });

  it('renders as disabled', async () => {
    const options = [
      { label: 'Option 1', value: '1' }
    ];
    const page = await newSpecPage({
      components: [VerdocsSelectInput],
      template: () => <verdocs-select-input disabled={true} options={options} />,
    });
    expect(page.root).toMatchSnapshot();
    const select = page.root.querySelector('select');
    expect(select?.hasAttribute('disabled')).toBe(true);
  });
});
