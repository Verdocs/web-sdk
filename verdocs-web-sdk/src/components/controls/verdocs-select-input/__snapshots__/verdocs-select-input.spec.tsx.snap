// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verdocs-select-input renders as disabled 1`] = `
<verdocs-select-input class="input-field">
  <label>
    <div></div>
    <select class="input-element" disabled="">
      <option value="1">
        Option 1
      </option>
    </select>
  </label>
</verdocs-select-input>
`;

exports[`verdocs-select-input renders with default props 1`] = `
<verdocs-select-input class="input-field">
  <label>
    <div></div>
    <select class="input-element"></select>
  </label>
</verdocs-select-input>
`;

exports[`verdocs-select-input renders with label, value, and options 1`] = `
<verdocs-select-input class="input-field">
  <label>
    <div class="input-label">
      Select:
    </div>
    <select class="input-element">
      <option value="1">
        Option 1
      </option>
      <option selected="" value="2">
        Option 2
      </option>
    </select>
  </label>
</verdocs-select-input>
`;
